lib/twitter_cldr/shared/properties_database.rb

```diff
@@ -0,0 +1,175 @@
+# encoding: UTF-8
+
+# Copyright 2012 Twitter, Inc
+# http://www.apache.org/licenses/LICENSE-2.0
+
+require 'set'
+
+module TwitterCldr
+  module Shared
+
+    class PropertiesDatabase
+      include TwitterCldr::Utils
+
+      DEFAULT_ROOT_PATH = File.join(
+        TwitterCldr::RESOURCES_DIR, 'unicode_data', 'properties'
+      )
+
+      attr_reader :root_path, :trie
+
+      def initialize(root_path = DEFAULT_ROOT_PATH)
+        @root_path = root_path
+        @trie = FileSystemTrie.new(root_path)
+      end
+
+      def store(property_name, property_value, data)
+        trie.add(key_for(property_name, property_value), data)
+      end
+
+      def code_points_for_property(property_name, property_value = nil)
+        key = key_for(property_name, property_value)
+        node = trie.get_node(key)
+
+        if node
+          if node.value
+            node.value
+          elsif name_indicates_value_prefix?(property_name)
+            concat_children(key)
+          end
```

Comment:
I'm actually not sure when the `else` case would ever happen, since the three possible cases are handled in the tests already (missing name, missing value, missing name and value). In all three cases the method returns an empty `RangeSet`. Just in case, I'll have the `else` branch here do the same thing, i.e. return an empty `RangeSet`.